server/modules/module_def.js

```diff
@@ -134,51 +134,53 @@ let loadModuleAtPath = function(path) {
  * The ModuleDef class contains all the information necessary to load & 
  * instantiate a module, including code location and config parameters.
  */
-class ModuleDef {
-  constructor(name, path, title, author, config) {
+class ModuleDef extends EventEmitter {
+  constructor(name, pathOrBaseModule, title, author, config) {
+    super();
     this.name = name;
-    this.path = path;
     this.config = config || {};
     this.title = title;
     this.author = author;
     
-    // The promise that when set, indicates the module is loaded.
-    this.loadPromise = null;
-    
     // The string source of the module.
     this.def_ = '';
-    
-    this.load_();
-  }
-  static register(def) {
-    assert(!(def.name in ModuleDef.modules), 'Def ' + def.name + ' already exists!');
-    ModuleDef.modules[def.name] = def;
+    if (pathOrBaseModule instanceof ModuleDef) {
+      // The promise that when set, indicates the module is loaded.
+      this.loadPromise = pathOrBaseModule.loadPromise.then(() => {
+        this.def_ = pathOrBaseModule.def_;
```

Comment:
Rather than inspecting the internals of the other module, why not just make the promise return the def?